site/app/controllers/SelfRejoinController.php

```diff
@@ -76,7 +127,7 @@ private function canRejoinCourse() {
             }
         }
 
-        $term_start_date = $this->core->getQueries()->getCurrentTermStartDate();
+        $term_start_date = $this->core->getQueries()->getTermStartDate($term, $user);
         // If never accessed course but today is within first two weeks of term, can re-add self.
```

Comment:
How is "never accessed course" enforced here? If it isn't, this line should be changed to avoid confusion